src/llmcompressor/entrypoints/oneshot.py

```diff
@@ -174,8 +174,160 @@ def apply_recipe_modifiers(
         session.finalize(**session_kwargs)
 
 
-def oneshot(**kwargs) -> PreTrainedModel:
-    one_shot = Oneshot(**kwargs)
+def oneshot(
+    # Required model parameters
+    model: str,
+    # Optional model parameters
+    distill_teacher: Optional[str] = None,
+    config_name: Optional[str] = None,
+    tokenizer: Optional[str] = None,
+    processor: Optional[str] = None,
```

Comment:
These can also be the loaded up classes. I think these are the best base classes to use? cc @kylesayrs 
```suggestion
    tokenizer: Optional[Union[str, transformers.PreTrainedTokenizerBase]] = None,
    processor: Optional[Union[str, transformers.ProcessorMixin]] = None,
```